From d3b7217f7305d5d4d85ffb6238e92ebe124b29ab Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Thu, 21 Mar 2024 11:40:48 +0100
Subject: [PATCH] media: verisilicon: h1: drop overfill when rotation enabled

Overfilled lines are seen when rotating image, so result is
worst than when doing nothing.
Reproduced with
gst-launch-1.0 videotestsrc ! video/x-raw, format=NV12, width=320, height=200 !  v4l2slh264enc rotation=90 ! v4l2slh264dec ! waylandsink
=> bad lines duplicated on the left of image
src(320x208)=>[rot=90]=>dst(200x320): overfill_r=0, overfill_b=8

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Change-Id: Ie303889760c97578b0fa3f72de4378b27ab995c9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368372
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/374019
---
 drivers/media/platform/verisilicon/hantro_h1.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/verisilicon/hantro_h1.c b/drivers/media/platform/verisilicon/hantro_h1.c
index 495622c6852c..4112cc942137 100644
--- a/drivers/media/platform/verisilicon/hantro_h1.c
+++ b/drivers/media/platform/verisilicon/hantro_h1.c
@@ -364,8 +364,9 @@ void hantro_h1_set_src_img_ctrl(struct hantro_dev *vpu, struct hantro_ctx *ctx)
 		overfill_r = ctx->src_fmt.width - ctx->dst_fmt.width;
 		overfill_b = ctx->src_fmt.height - ctx->dst_fmt.height;
 	} else {
-		overfill_r = ctx->src_fmt.width - ctx->dst_fmt.height;
-		overfill_b = ctx->src_fmt.height - ctx->dst_fmt.width;
+		/* FIXME Overfilled lines are seen when rotating image */
+		overfill_r = 0;
+		overfill_b = 0;
 	}
 
 	reg = H1_REG_IN_IMG_CTRL_ROW_LEN(ctx->src_fmt.width)
-- 
2.39.5

